components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/dashboard/ArtifactUpdateListener.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.initializer.dashboard;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+/**
+ * Manages updated artifacts in micro integrator using the management api to notify the dashboard.
+ */
+public class ArtifactUpdateListener {
+    public static final String PROXY_SERVICE_TRACE = "proxy_service_trace";
+    public static final String API_TRACE = "api_trace";
+    public static final String INBOUND_ENDPOINT_TRACE = "inbound_endpoint_trace";
+    public static final String SEQUENCE_TEMPLATE_TRACE = "sequence_template_trace";
+    public static final String ENDPOINT_TRACE = "endpoint_trace";
+
+    public static final String PROXY_SERVICES = "proxy-services";
+    public static final String APIS = "apis";
+    public static final String INBOUND_ENDPOINTS = "inbound-endpoints";
+    public static final String SEQUENCE_TEMPLATE = "templates_sequence";
+    public static final String SEQUENCES = "sequences";
+    public static final String ENDPOINTS = "endpoints";
+
+    private ArtifactUpdateListener() {
+
+    }
+
+    private static JsonArray updatedArtifacts = new JsonArray();
+    
+    public static void addToUpdatedArtifactsQueue(String updateType, String name) {
+
+        JsonObject updatedArtifact = getUpdatedArtifact(updateType, name);
+        updatedArtifacts.add(updatedArtifact);
+    }
+
+    public static JsonArray getUpdatedArtifacts() {
+        return updatedArtifacts;
+    }
+
+    public static void removeFromUpdatedArtifactQueue(int artifactsSize) {
+        for (int i = 0; i < artifactsSize; i++) {
+            updatedArtifacts.remove(0);
+        }
+    }
+
+    private static JsonObject getUpdatedArtifact(String updateType, String name) {
+        JsonObject updatedArtifact = new JsonObject();
+        updatedArtifact.addProperty("type", getType(updateType));
+        updatedArtifact.addProperty("name", name);
+        return updatedArtifact;
+    }
+
+    private static String getType(String updateType) {
```

Comment:
Shall we get the artifactType from the handleTracing method above and keep this class generic so that we don't have to change this class for every artifact we support update in future?

Reply:
Updated the logic so that the artifact type will be send from handleTracing[1]
[1] https://github.com/wso2/micro-integrator/blob/2c968f639e635be0eb13cdd45b252fe000759819/components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java#L131